cmd/kubelet/app/server.go

```diff
@@ -439,13 +454,26 @@ func run(s *options.KubeletServer, kcfg *KubeletConfig) (err error) {
 	return nil
 }
 
+// getNodeName returns the node name according to the cloud provider
+// if cloud provider is specified. Otherwise, returns the hostname of the node.
+func getNodeName(cloud cloudprovider.Interface, hostname string) (string, error) {
```

Comment:
having this factored out is good, but we should use it in RunKubelet as well or we'll drift from how the node name is computed for real. pushed a commit to https://github.com/liggitt/kubernetes/commits/tls-bootstrap that unifies those two paths (and fixes the NPE at https://github.com/kubernetes/kubernetes/pull/30922/files/4120179db5ce6c3fadc872a268347f5be038f98f#r75416356)